doc/source/whatsnew/v0.22.0.txt

```diff
@@ -36,7 +36,7 @@ Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 - :func:`Series.fillna` now raises a ``TypeError`` instead of a ``ValueError`` when passed a list, tuple or DataFrame as a ``value`` (:issue:`18293`)
--
+- The default NA value for :class:`UInt64Index` has changed from 0 to ``NaN`` (:issue:`18398`)
```

Comment:
What user visible change has this? 
Now it sounds a bit vague

Reply:
It impacts things that  mask with NA under the hood, such as `.where` 

Previous behavior:
```python
In [3]: idx = pd.UInt64Index(range(5))

In [4]: idx
Out[4]: UInt64Index([0, 1, 2, 3, 4], dtype='uint64')

In [5]: idx.where(idx > 3)
Out[5]: UInt64Index([0, 0, 0, 0, 4], dtype='uint64')
```

New behavior:
```python
In [3]: idx = pd.UInt64Index(range(5))

In [4]: idx.where(idx > 3)
Out[4]: Float64Index([nan, nan, nan, nan, 4.0], dtype='float64')
```

Updating the whatsnew to make this more clear.